### ISSUE TITLE
Replace dict.has_key() with the 'in' keyword, for twisted/test

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8365]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8365|
|Type|enhancement|
|Created|2016-05-26 17:59:23Z|
|Branch|https://github.com/twisted/twisted/tree/8365-rodrigc-has_key7|

In PEP 0290, has_key() was deprecated in Python 2.2 and higher: 
* https://www.python.org/dev/peps/pep-0290/#testing-dictionary-membership
* https://docs.python.org/2.2/whatsnew/node4.html

In Python 3, dict.has_key() was removed:
* https://docs.python.org/3.0/whatsnew/3.0.html#builtins

PEP 0290 says that for testing dictionary membership, this:

```
if d.has_key(k):
```

should be changed to this:

```
if k in d:
```

<details><summary>Searchable metadata</summary>

```
trac-id__8365 8365
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8365_rodrigc_has_key7 8365-rodrigc-has_key7
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464285563414140 1464285563414140
changetime__1465460684258024 1465460684258024
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/81

This was pulled out from [ticket:4053]

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Changes looks good. Thanks!

Waiting for test results.

I have added a topfile and also updated the tests to use 'assertIn' instead of just assert

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 84c674d2923aa3a57ff1f3c955591f669fdb81c9
```
#!CommitTicketReference repository="" revision="84c674d2923aa3a57ff1f3c955591f669fdb81c9"
Merge 8365-rodrigc-has_key7: Replace dict.has_key() in twisted.test.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8365
```
